src/phases/query/QueryStats.tmpl

```diff
@@ -0,0 +1,230 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This is a template library of actors for QueryStats workloads.
+
+  Actors (in the order they should appear in the workload):
+    ClearCollection: (requires MaxPhase, Phase) deletes the working collection, should only need to be run once.
+    InsertData: (requires MaxPhase, Phase) populates the working collection, should only need to be run once.
+    CacheSizeActor: (requires MaxPhase, Phases, CacheSizeMb) changes the QueryStats cache size.
+    RateLimitActor: (requires MaxPhase, Phases, RateLimitHz) changes the QueryStates rate limit. Should be set to -1 before each QueryStats cache prepopulation to speed things up.
+    QuiesceActor: (requires MaxPhase, Phases) ensures that actors quiesce before continuing.
+    PrepopulateActor: (requires MaxPhase, Phases, CacheSizeMb) prepopulates the QueryStats cache to ensure that it is filled with similarish shapes before every workload execution.
+    FindCommandActor: (requires MaxPhase, Phase, CacheSizeMb, RateLimitHz, Payload; optional Suffix) runs find commands to generate query stats.
+    AggregateActor: (requires MaxPhase, Phase, CacheSizeMb, RateLimitHz; optional Suffix, QueryStatsParameters) runs $queryStats.
+
+  Parameters:
+    # Required parameters.
```

Comment:
I'm a little confused by some of this setup reading through for the first time - above you say that certain actors have different required parameters - here it appears that all of these parameters are required and that we'll ben instantiating the template exactly once?

I think maybe just delete this comment? Or if you have a better understanding of it you could summarize, do that :D

Reply:
Oh -- some of the required parameters are simply not used in some actors. Genny's idea of "required" is simply that the default value is something that is unlikely to be accepted, so that the runtime will error unless that value is overridden. I can explain that further in the template, sure.